feat: update WCP with assembly implementation

DavePearce opened the pull request:
The purpose of this PR is to replace the existing `wcp` module with an assembly implementation.  This requires first breaking the dependency of `rlputils` on the column `wcp.CT_MAX` (as this column will no longer exist).  To do this, we make use of a secondary function `maxlog()` which computes the necessary information instead (i.e. it calculates the value previously determined by `wcp.CT_MAX`).

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Simplifies WCP by deleting `WcpCall` and reducing `WcpOperation` to a single-row trace while updating CI to Go 1.24.6 and go-corset v1.1.28.
> 
> - **Arithmetization / WCP**:
>   - Simplify `WcpOperation` tracing to a single row via `trace.inst(...).arg1(...).arg2(...).res(...)` and `computeLineCount()` now returns `1`.
>   - Remove per-byte/bit logic, counters, and `ctMax`-driven paths; delete `WcpCall` class.
>   - Keep opcode result computation; comparator unchanged aside from a clarifying comment.
> - **CI**:
>   - Update `.github/actions/setup-go-corset/action.yml` to Go `1.24.6` and `go-corset` `v1.1.28`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ca39d093cab7ea1829246f6b6e87f8b19558e66f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

DavePearce committed:
update linea-constraints

This updates linea-constraints to include the maxlog() function which
breaks the dependency of rlputils on wcp.CT_MAX.

DavePearce committed:
update Wcp module for assembly implementation

This makes various changes to the Wcp module to support the assembly
implementation.  Mostly, this involves removing column calculations
which are no longer required.

This also updates the constraints to utilise the maxlog function within
rlputils, and updates go-corset to v1.1.28 (which is required for the
defcall syntax).

letypequividelespoubelles commented:
`update WCP with assembly implementation` is not done in this PR, it's only chnging the call from RLP_UTILS here :)

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java:
```diff
@@ -157,97 +83,15 @@ private Bytes16 calculateAdj(boolean cmp, BigInteger arg1, BigInteger arg2) {
-    final boolean oli = isOli();
-    final boolean vli = isVli();
+    // Calculate result
+    final boolean resLo = calculateResult(wcpInst, arg1, arg2);
```
better name it resBool, or just res, there are no high/lo part of the (bin) result.

DavePearce replied:
Done

letypequividelespoubelles reviewed (approved):
LGTM

DavePearce committed:
rename resLo => res

This just renames the variable to better reflect what it is.
Specifically, it is not the "low" part of the result. Rather, it is the
result.
